app/lib/router/router.dart

```diff
@@ -70,14 +70,27 @@ Future<String?> forwardRedirect(
       final deviceId = state.uri.queryParameters['deviceId']
           .expect('query params should contain device id');
       client = await acterSdk.getClientWithDeviceId(deviceId, true);
-      // ignore: use_build_context_synchronously
-      final ref = ProviderScope.containerOf(context);
-      // ensure we have selected the right client
-      ref.invalidate(clientProvider);
     } catch (e, s) {
-      _log.severe('Client not found', e, s);
-      return null;
+      _log.severe('Specified Client not found', e, s);
```

Comment:
this update on the forwarding code now falls back to an existing client if the deviceId failed to materialize and only if that isn't possible, forwards to the intro/login screens.

Before this would return `null` here some times leading to an empty screen being rendered.